### PR TITLE
Statement generation patch

### PR DESCRIPTION
**Pull request by cip999**
Added support for "Interaction" section in problem statement.

### CODE REVIEW

File: p2d/parse_polygon_package.py [hunk 1]
```diff
@@ -27,7 +27,7 @@ def parse_samples_explanations(notes):
                 exit(1)
             assert(test_id != -1)
             assert(test_id not in explanations)
-            curr = curr[0].upper() + curr[1:]  # Capitalize first letter.
+            curr = curr[0].upper() + curr[1:]   # Capitalize first letter.
```
Comment (dario2994) [inline p2d/parse_polygon_package.py:30]: Unless there's a reason, please undo this change.

File: p2d/tex_utilities.py [hunk 2]
```diff
@@ -77,6 +85,7 @@ def generate_statement_tex(problem, tex_dir):
         'LEGEND': problem['statement']['legend'],
         'INPUT': problem['statement']['input'],
         'OUTPUT': problem['statement']['output'],
```
Comment (dario2994) [inline p2d/tex_utilities.py:87]: Also INPUT and OUTPUT may be None, right? If that is the case, treat them like you treat INTERACTION

File: p2d/tex_utilities.py [hunk 3]
```diff
@@ -77,6 +85,7 @@ def generate_statement_tex(problem, tex_dir):
...
+        'INTERACTION': '' if problem['statement']['interaction'] is None else problem['statement']['interaction'],
```
Comment (dario2994) [inline p2d/tex_utilities.py:88]: It is more appropriate to move this (None -> '') to parse_polygon_package because this is "morally" about parsing the package into a nice object.

**Review by dario2994 — CHANGES_REQUESTED** [review]
It looks good. After you fix these minor issues, it can be merged.

File: p2d/parse_polygon_package.py (hunk 1 above)
Comment (cip999) [reply p2d/parse_polygon_package.py:30]: Nope, it was accidental, I'll undo it.

File: p2d/tex_utilities.py (hunk 2 above)
Comment (cip999) [reply p2d/tex_utilities.py:87]: From what I've seen in polygon packages, they can only be a (possibly empty) string. But a sanity check for null is not a bad idea, just in case.
